src/main/java/com/google/sps/servlets/PhotoHandlerServlet.java

```diff
@@ -0,0 +1,172 @@
+
+/**
+ * When the user submits the form for image uploading, Blobstore processes the file upload and forwards
+ * the request to this servlet, which returns the product shopping results for the respective photo,
+ * in JSON format.
+ */
+@WebServlet("/handle-photo")
+public class PhotoHandlerServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the BlobKey that points to the image uploaded by the user.
+    BlobKey blobKey = getBlobKey(request, "photo");
+
+    // Send an error if the user did not upload a file.
+    if (blobKey == null) {
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Client must upload an image file.");
+      return;
+    }
+
+    // Get the photo category (i.e. product, shopping-list or barcode) entered by the user.
+    if (request.getParameter("photo-category").isEmpty()) {
+      response.sendError(
+          HttpServletResponse.SC_BAD_REQUEST, "Client must select a photo category when submitting the form.");
+      return;
+    }
+    String photoCategory = request.getParameter("photo-category");
+
+    // Get the image the user uploaded as bytes.
+    byte[] imageBytes = getBlobBytes(blobKey);
+
+    // TO ADD: Based on the {@code photoCategory}, call methods from the photo detection classes, 
+    // passing {@code imageBytes} as argument. These methods detect the image content, build 
+    // the shopping query and call GoogleShoppingQuerier to return the extracted products.
+
+    // Build the shopping query input - set {@code shoppingQuery}, {@code language} and 
+    // {@code maxResultsNumber} to hard-coded values for now.
+    // TO DO: Let the user choose {@code language} and {@code maxResultsNumber}.
+    String shoppingQuery = getQuery(request.getParameter("photo-category"));
+    ShoppingQueryInput input = 
+        new ShoppingQueryInput.Builder(shoppingQuery).language("en").maxResultsNumber(21).build();
+
+    // Initialize the Google Shopping querier.
+    GoogleShoppingQuerier querier = new GoogleShoppingQuerier();
+    
+    response.setContentType("application/json;");
+    
+    List<Product> shoppingQuerierResults = new ArrayList<>();
+    try {
+      shoppingQuerierResults = querier.query(input);
+    } catch(IllegalArgumentException | ShoppingQuerierConnectionException | IOException exception) {
+      response.sendError(SC_INTERNAL_SERVER_ERROR, exception.getMessage());
+    }
+     
+    // Convert products List into a JSON string using Gson library and
+    // send the JSON as the response.
+    Gson gson = new Gson();
+
+    response.getWriter().println(gson.toJson(shoppingQuerierResults));
+  }
+  
+  private String getQuery(String photoCategory) {
+    switch (photoCategory) {
+      case "product":
+        return "Fountain pen";
+      case "shopping-list":
+        return "Fuzzy socks";
+      case "barcode":
+        return "Running shoes";
+      default:
```

Comment:
This hard-coded method is taken from a previous PR and I left it with your exact implementation, considering that we are actually going to delete this soon, when adding the photo detection methods. Actually, I think I should delete this in this PR

Reply:
I am not sure who added the default there but its still good to change it anyways now that we found it might raise bugs.

And I don't think we will delete this. 
We are still using `photo-category`.
We are just going to use our product detection and text detection class' objects and their methods instead of hard coded queries right?
We will still use switch case unless we change how we are receiving the image from user. Which I believe we are going to do after we merge our MVP in master and not now.

Reply:
Indeed, now that the misunderstanding created by the workflow scheme has been resolved, I readded the method and also handled the error